src/gromacs/nblib/topology.cpp

```diff
@@ -182,8 +298,41 @@ gmx::ListOfLists<int> TopologyBuilder::createExclusionsListOfLists() const
     return exclusionsListOfListsGlobal;
 }
 
+Topology::InteractionData TopologyBuilder::createInteractionData(const detail::ParticleSequencer& particleSequencer)
+{
+    auto create = [](auto& output, const auto& molecules, const auto& sequencer) {
```

Comment:
This is impossible to understand and so will be impossible to maintain. What is output? How do I know that it has members `bondInstance` and `indices`?

Reply:
Yeah, this is a bit much for a lambda. I think you could explicitly type the Topology::InteractionData (and other parameters) or make this a template free function (with SFINAE checking) if you need reusability.

Reply:
I can't explicitly type the lambda because it gets called with the _elements_ of Topology::InteractionData as arguments. I have restructured and simplified the lambda to make it more evident that its body would
be the content of a traditional for-loop over the elements of the InteractionData tuple if that were possible.